File: lib/config/admin_config.dart
/// Admin paneline erişebilen Firebase Auth UID listesi.
/// Firestore rules ile senkron tutun (firestore.rules → isAdmin).
import 'package:firebase_auth/firebase_auth.dart';

const adminUids = <String>[
  // falora.admin@falora.app
  'l2vjDZXUbcOifjumq1q7HTubFxu2',
  // henryarthur.rr@gmail.com
  'o0AyRibzJFfNb6BNYxUweM4RqAs2',
  // admin.35@tombikteyze.app
  'LmcMFwbp3eaLRrFRD2IURLW3reg1',
  // admin@tombikteyze.app
  'PcyiSwnAIWhkwlpsJ88fHYMFGJW2',
];

/// UID listesine ek güvenlik ağı.
const adminEmails = <String>[
  'falora.admin@falora.app',
  'henryarthur.rr@gmail.com',
  'admin.35@tombikteyze.app',
  'admin@tombikteyze.app',
];

bool isAdminUser(String uid, {String? email}) {
  if (adminUids.contains(uid)) return true;
  // Yalnızca Firebase Auth e-postası ile eşle (Firestore profili değil).
  final normalized = email?.trim().toLowerCase();
  if (normalized != null &&
      normalized.isNotEmpty &&
      adminEmails.contains(normalized)) {
    return true;
  }
  return false;
}

/// Admin yönlendirmesi için Firebase Auth oturumunu kullan.
bool isCurrentFirebaseUserAdmin() {
  final fbUser = FirebaseAuth.instance.currentUser;
  if (fbUser == null) return false;
  return isAdminUser(fbUser.uid, email: fbUser.email);
}

File: backend/admin_config.js
/** Firestore rules ve lib/config/admin_config.dart ile senkron tutun. */
const adminUids = [
  'l2vjDZXUbcOifjumq1q7HTubFxu2', // falora.admin@falora.app
  'o0AyRibzJFfNb6BNYxUweM4RqAs2', // henryarthur.rr@gmail.com
  'LmcMFwbp3eaLRrFRD2IURLW3reg1', // admin.35@tombikteyze.app
  'PcyiSwnAIWhkwlpsJ88fHYMFGJW2', // admin@tombikteyze.app
];

const adminEmails = [
  'falora.admin@falora.app',
  'henryarthur.rr@gmail.com',
  'admin.35@tombikteyze.app',
  'admin@tombikteyze.app',
];

function isAdminUid(uid) {
  return typeof uid === 'string' && adminUids.includes(uid);
}

function isAdminEmail(email) {
  if (typeof email !== 'string') return false;
  const normalized = email.trim().toLowerCase();
  return normalized.length > 0 && adminEmails.includes(normalized);
}

function isAdminUser(uid, email) {
  return isAdminUid(uid) || isAdminEmail(email);
}

module.exports = {
  adminUids,
  adminEmails,
  isAdminUid,
  isAdminEmail,
  isAdminUser,
};
